lib/membrane_wav/parser.ex

```diff
@@ -90,11 +90,11 @@ defmodule Membrane.WAV.Parser do
     mode: :pull,
     availability: :always,
     demand_unit: :bytes,
-    caps: Caps
+    caps: :any
 
   @impl true
-  def handle_init(state) do
-    state = Map.merge(state, %{stage: :init})
+  def handle_init(options) do
+    state = Map.merge(options, %{stage: :init})
```

Comment:
One more nitpick: it's still kind of struct now as you keep the original map. A bit of context now: structs are implemented as maps with a special `__struct__` field so essentially matching on  `%SomeStruct{}` is the same as `%{__struct__: SomeStruct}` (except for additional compile time checks). Now, in this case the result is:
```
state: %{__struct__: Membrane.WAV.Parser, frames_per_buffer: 2048, stage: :init}
```
it's not printed as struct because it's no longer valid `Parser` struct, but the key `__struct__` remains

Additionally, when adding only one key you can use `Map.put/3`
```suggestion
    state = options |> Map.from_struct() |> Map.put(:stage, :init)
```